db/log_reader.h

```diff
@@ -147,6 +152,17 @@ class Reader {
   // which log number this is
   uint64_t const log_number_;
 
+  // See `Optinos::track_and_verify_wals`
+  bool track_and_verify_wals_;
+  // Below variables are used for WAL verification
+  // TODO(hx235): to revise `stop_replay_for_corruption_` in `LogReader` since
+  // we have `predecessor_wal_info_` to verify against the `PredecessorWALInfo`
+  // recorded in current WAL. If there is no WAL hole, we can revise
+  // `stop_replay_for_corruption_` to be false.
+  bool stop_replay_for_corruption_;
```

Comment:
This means we can do MaybeReviseStopReplayForCorruption() https://github.com/facebook/rocksdb/blob/d957e1a33af7201b6ac3601baa4fd65b0958a25f/db/db_impl/db_impl_open.cc#L1518-L1521 in a better way that is compatible with disable WAL, file ingestion when WAL is recycled. So we can fix the issue mentioned https://github.com/facebook/rocksdb/pull/12918 and re-enable ecycle_log_file_num (wal recycling)